lib/ddtrace/profiling/collectors/stack.rb

```diff
@@ -187,6 +187,9 @@ def build_backtrace_location(_id, base_label, lineno, path)
 
         private
 
+        # FIXME: Threads started with `start`/`fork` can wrongly trigger this warning, because unlike with Thread#new
+        # we can't distinguish between missing instrumentation and "thread was just started and hasn't had time to
+        # run update_native_ids"
```

Comment:
Ugh, over the lifetime of an application this will likely get triggered a few times.

Would it make sense to give a thread one opportunity to fulfill `@warned_about_missing_cpu_time_instrumentation`, something like setting a sentinel value in the thread object that tracks "we've given you one chance to run update_native_ids, but next time I check, I'll warn if you didn't". But this suggestion has the side-effect of missing the warning on short-lived threads.

What do you think?

Reply:
Agreed. It's a shame that it ends up making the detection miss a few cases, but hopefully we're mostly done with the whack-a-mole game so it's not going to come up often.

I'll wait until #1310 gets merged to add this change, as there's already quite a bit of branch dependencies around, and after that I'll unmark as draft.